tests/receivers/envoy/bundled_k8s_test.go

```diff
@@ -124,39 +125,46 @@ func TestEnvoyK8sObserver(t *testing.T) {
 			return
 		}
-
-		i++
-		err := pmetrictest.CompareMetrics(expected, sink.AllMetrics()[len(sink.AllMetrics())-1],
-			pmetrictest.IgnoreResourceAttributeValue("service.instance.id"),
-			pmetrictest.IgnoreResourceAttributeValue("net.host.port"),
-			pmetrictest.IgnoreResourceAttributeValue("net.host.name"),
-			pmetrictest.IgnoreResourceAttributeValue("server.address"),
-			pmetrictest.IgnoreResourceAttributeValue("container.name"),
-			pmetrictest.IgnoreResourceAttributeValue("server.port"),
-			pmetrictest.IgnoreResourceAttributeValue("service.name"),
-			pmetrictest.IgnoreResourceAttributeValue("service_instance_id"),
-			pmetrictest.IgnoreResourceAttributeValue("service_version"),
-			pmetrictest.IgnoreResourceAttributeValue("discovery.endpoint.id"),
-			pmetrictest.IgnoreMetricAttributeValue("service_version"),
-			pmetrictest.IgnoreMetricAttributeValue("service_instance_id"),
-			pmetrictest.IgnoreResourceAttributeValue("server.address"),
-			pmetrictest.IgnoreResourceAttributeValue("k8s.pod.name"),
-			pmetrictest.IgnoreResourceAttributeValue("k8s.pod.uid"),
-			pmetrictest.IgnoreTimestamp(),
-			pmetrictest.IgnoreStartTimestamp(),
-			pmetrictest.IgnoreMetricDataPointsOrder(),
-			pmetrictest.IgnoreScopeMetricsOrder(),
-			pmetrictest.IgnoreScopeVersion(),
-			pmetrictest.IgnoreResourceMetricsOrder(),
-			pmetrictest.IgnoreMetricsOrder(),
-			pmetrictest.IgnoreMetricValues(),
-		)
-		assert.NoError(tt, err)
+		var errs error
+		newIndex := len(sink.AllMetrics())
+		for i := index; i < newIndex; i++ {
+			err := pmetrictest.CompareMetrics(expected, m,
+				pmetrictest.IgnoreResourceAttributeValue("service.instance.id"),
+				pmetrictest.IgnoreResourceAttributeValue("net.host.port"),
+				pmetrictest.IgnoreResourceAttributeValue("net.host.name"),
+				pmetrictest.IgnoreResourceAttributeValue("server.address"),
+				pmetrictest.IgnoreResourceAttributeValue("container.name"),
+				pmetrictest.IgnoreResourceAttributeValue("server.port"),
+				pmetrictest.IgnoreResourceAttributeValue("service.name"),
+				pmetrictest.IgnoreResourceAttributeValue("service_instance_id"),
+				pmetrictest.IgnoreResourceAttributeValue("service_version"),
+				pmetrictest.IgnoreResourceAttributeValue("discovery.endpoint.id"),
+				pmetrictest.IgnoreMetricAttributeValue("service_version"),
+				pmetrictest.IgnoreMetricAttributeValue("service_instance_id"),
+				pmetrictest.IgnoreResourceAttributeValue("server.address"),
+				pmetrictest.IgnoreResourceAttributeValue("k8s.pod.name"),
+				pmetrictest.IgnoreResourceAttributeValue("k8s.pod.uid"),
+				pmetrictest.IgnoreTimestamp(),
+				pmetrictest.IgnoreStartTimestamp(),
+				pmetrictest.IgnoreMetricDataPointsOrder(),
+				pmetrictest.IgnoreScopeMetricsOrder(),
+				pmetrictest.IgnoreScopeVersion(),
+				pmetrictest.IgnoreResourceMetricsOrder(),
+				pmetrictest.IgnoreMetricsOrder(),
+				pmetrictest.IgnoreMetricValues(),
+			)
+			if err == nil {
+				return
+			}
+			errs = errors.Join(errs, err)
```

Comment:
I'm afraid multiple errors aggregated within a second can be hard to read. But we'll see

Reply:
Actually, not sure what the value of joining here, if we only join the errors from metrics received in the last second. I'd suggest throwing the last one. At least it'll be readable

Reply:
no problem, changed.